library/Icingadb/Web/Controller.php

```diff
@@ -349,19 +349,47 @@ public function createViewModeSwitcher()
             $viewModeSwitcher->setDefaultViewMode($viewMode);
         }
 
-        $viewModeSwitcher->populate(['view' => $this->params->shift($viewModeSwitcher->getViewModeParam())]);
+        $viewModeSwitcher->populate([
+            'view' => $this->params->shift($viewModeSwitcher->getViewModeParam())
+        ]);
 
-        $viewModeSwitcher->on(ViewModeSwitcher::ON_SUCCESS, function (ViewModeSwitcher $viewModeSwitcher) use ($prefs) {
-            $viewMode = $viewModeSwitcher->getValue($viewModeSwitcher->getViewModeParam());
+        if ($viewModeSwitcher->getDefaultViewMode() === 'minimal') {
+            if ($limitControl->getLimit() === $limitControl->getDefaultLimit()) {
+                $limitControl->setDefaultLimit($limitControl->getDefaultLimit() * 2);
+            }
 
-            $icingadbPrefs = $prefs->icingadb ?: [];
-            $icingadbPrefs['view_mode'] = $viewMode;
-            $prefs->icingadb = $icingadbPrefs;
+            $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() * 2);
+        }
 
-            $this->redirectNow(Url::fromRequest()->setParam($viewModeSwitcher->getViewModeParam(), $viewMode));
-        });
+        $viewModeSwitcher->on(
+            ViewModeSwitcher::ON_SUCCESS,
+            function (ViewModeSwitcher $viewModeSwitcher) use ($prefs, &$paginationControl) {
+                $viewMode = $viewModeSwitcher->getValue($viewModeSwitcher->getViewModeParam());
+                $icingadbPrefs = $prefs->icingadb ?: [];
+                $icingadbPrefs['view_mode'] = $viewMode;
+                $prefs->icingadb = $icingadbPrefs;
+                $currentPage = $paginationControl->getCurrentPageNumber();
+
+                if ($viewMode === 'minimal') {
+                    $limit = $paginationControl->getLimit();
+                    $currentPage = ((($paginationControl->getCurrentPageNumber() * $limit) - $limit) / ($limit * 2) + 1);
+
+                    $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() * 2);
+                } elseif ($viewModeSwitcher->getDefaultViewMode() === 'minimal') {
+                    $limit = $paginationControl->getLimit();
+                    $currentPage = ((($paginationControl->getCurrentPageNumber() * $limit) - $limit) / ($limit / 2) + 1);
+
+                    $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() / 2);
+                }
+
+                $this->redirectNow(Url::fromRequest()->setParams([
+                    $viewModeSwitcher->getViewModeParam()   => $viewMode,
+                    $paginationControl->getPageParam()      => $currentPage
```

Comment:
Please don't add the `page` param if it's value is `1`. That's the default anyway and I'd like to avoid adding the `page` param although it has not been there previously.